code/extensions/CDNFile.php

```diff
@@ -209,6 +209,10 @@ public function onAfterWrite() {
                 $pathBefore = $changedFields['Filename']['before'];
                 $pathAfter = $changedFields['Filename']['after'];
 
+                if($this->owner instanceof Folder) {
+                    $this->updateChildFolderLinks($this->owner->Children(), $pathBefore, $pathAfter);
```

Comment:
And shouldn't this only execute within the if () statement below ($pathBefore != $pathAfter)